File: BarberApp_Nhom6_Be/app/schemas/stylist.py
from pydantic import BaseModel
from typing import Optional, List

# 🟢 Schema tạo stylist
class StylistCreate(BaseModel):
    shop_id: int
    name: str
    bio: Optional[str] = None
    avatar_url: Optional[str] = None
    is_active: Optional[bool] = True
    service_ids: Optional[List[int]] = None  # danh sách dịch vụ stylist cung cấp

# 🟡 Schema cập nhật stylist
class StylistUpdate(BaseModel):
    name: Optional[str] = None
    bio: Optional[str] = None
    avatar_url: Optional[str] = None
    is_active: Optional[bool] = None
    service_ids: Optional[List[int]] = None

# 🔵 Schema trả dữ liệu ra client
class StylistOut(BaseModel):
    id: int
    shop_id: int
    name: str
    bio: Optional[str]
    avatar_url: Optional[str]
    is_active: bool

    class Config:
        from_attributes = True

File: BarberApp_Nhom6_Fe/barberapp_nhom6_fe/lib/models/stylist.dart
class Stylist {
  final int id;
  final int shopId;
  final String name;
  final String? bio;
  final String? avatarUrl;
  final bool isActive;
  final List<int>? serviceIds; // <-- Thêm

  Stylist({
    required this.id,
    required this.shopId,
    required this.name,
    this.bio,
    this.avatarUrl,
    required this.isActive,
    this.serviceIds,
  });

  factory Stylist.fromJson(Map<String, dynamic> json) {
    return Stylist(
      id: json['id'] as int,
      shopId: json['shop_id'] as int,
      name: json['name'] ?? '',
      bio: json['bio'],
      avatarUrl: json['avatar_url'],
      isActive: json['is_active'] ?? true,
      serviceIds: json['service_ids'] != null
          ? List<int>.from(json['service_ids'])
          : null,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'shop_id': shopId,
    'name': name,
    'bio': bio,
    'avatar_url': avatarUrl,
    'is_active': isActive,
    'service_ids': serviceIds,
  };
}
